### PR TITLE
EZAF-4409: Simplify/Improve Ray Tutorials

### PR DESCRIPTION
**Pull request by sercanCyberVision**
**MOTIVATION**
For tutorials to be effective, clarity and conciseness are key. Equally important, each tutorial should serve a distinct purpose, be it a comparative analysis, demonstrating a specific configuration, or illustrating a unique use case.

**SOLUTION**
This PR introduces a comprehensive restructuring to enhance tutorial organization and purposefulness:
1. Consolidation of all materials into a single primary folder named `Ray`.
2. Introduction of two subfolders: `GPU` and `CPU`, to distinguish between computational types.
3. Inclusion of two CPU-focused examples and one GPU-focused example, each equipped with its dedicated `README.md` file for detailed instructions and insightful explanations.
4. Implementation of a consistent structural template across all tutorials to facilitate user navigation and understanding.

**TESTS**
- The restructured tutorial collection has been meticulously organized as depicted below:
image

- The Fibonacci example aims to showcase the efficiency of Ray in comparison to local Python execution under different workloads, as demonstrated in the following result comparison:
result-comparison

- The Hyperparameter Tuning example illustrates the utilization of Ray Tune for optimizing model configurations to achieve the best possible settings:
hyperparameter-result

- The GPU tutorial details the steps for enabling GPU support within Ray and confirms the execution of jobs with GPU acceleration:
result

@Manish-2004 please execute the jobs on your end to make sure everything works fine.

### CODE REVIEW

**Comment by ask664** [comment]
after merging this PR we have to rebuild ezua-tutorial image and raise PR on ezaf and hefo repositories.

File: Data-Science/Ray/Ray-CPU/Fibonacci/README.md
Comment (rashminau) [inline Data-Science/Ray/Ray-CPU/Fibonacci/README.md:9]: I think for bullet point we need to specify which exact directory and what necessary files are required in that directory for the Ray job.

File: Data-Science/Ray/Ray-CPU/Hyperparameter/README.md
Comment (rashminau) [inline Data-Science/Ray/Ray-CPU/Hyperparameter/README.md:9]: Same for this one too. I think we need to specify the exact working directory and exact files that are necessary to submit this Ray job.

File: Data-Science/Ray/Ray-CPU/Fibonacci/README.md
Comment (sercanCyberVision) [reply Data-Science/Ray/Ray-CPU/Fibonacci/README.md:9]: Actually there is no exact directory, I was trying to say that it is better to use a dedicated directory containing only the necessary files for each job, because each example pushes its current dir to Ray cluster as in below:
```
    job_id = c.submit_job(
        entrypoint=f"python ray-fibonacci-example.py {n}",
        runtime_env={
            "working_dir": "./",     <==== This transfers all files in the current directory to the Ray cluster.
        },
        entrypoint_num_cpus=3
    )
```

I notice that some people submit jobs from common/root directory that leads pushing too many unrelated files to Ray cluster. This causes object-memory issues.

Maybe my wording did not reflect this properly, can you help with better explanation?

Comment (rashminau) [reply Data-Science/Ray/Ray-CPU/Fibonacci/README.md:9]: Thanks, Sercan! Will it be technically accurate to suggest users to run the tutorials from the /<username> directory? 

Maybe something like this, "To ensure optimal performance, use dedicated directories containing only the essential files needed for that job submission as a working directory. For example, If you do not see the Fibonacci folder in the <username> directory, copy the folder from the /shared/ezua-tutorials/Data-Science/Ray directory into the <username> directory. The /shared directory is accessible to all users. Editing or running examples from the /shared directory is not advised. The <username> directory is specific to you and cannot be accessed by other users."

I think it is okay if we want to keep instructions short here and directly add these detailed explanation in the documentation site for Ray tutorials too.

Comment (sercanCyberVision) [reply Data-Science/Ray/Ray-CPU/Fibonacci/README.md]: Thank you for correcting this very detailed. I have used the short version `To ensure optimal performance, use dedicated directories containing only the essential files needed for that job submission as a working directory` in readme files. As you mentioned, we can use detailed version in our documentation site.

Is this good for you?

**Comment by sercanCyberVision** [comment]
Please do not merge this until @Manish-2004 confirms that he is able to run everything smoothly on his end too.

**Comment by ask664** [comment]
@Manish-2004 any update on this?

**Review by Manish-2004 — APPROVED** [review]

**Review by ask664 — APPROVED** [review]

File: Data-Science/Ray/Ray-CPU/Matrix_Multiplication/README.md
Comment (sercanCyberVision) [inline Data-Science/Ray/Ray-CPU/Matrix_Multiplication/README.md:6]: Why do we need to choose Tensorflow notebook? Anything missing with Data Science notebook?

File: Data-Science/Ray/Ray-CPU/Matrix_Multiplication/README.md
Comment (sercanCyberVision) [inline Data-Science/Ray/Ray-CPU/Matrix_Multiplication/README.md]: Line above say `Below is a screenshot showing ...`, can you add a screenshot here that shows the final result? Please keep the file in `resources` directory

Comment (sercanCyberVision) [reply Data-Science/Ray/Ray-CPU/Matrix_Multiplication/README.md]: Ohh! This is done with the next commit, I missed it. Please ignore this one

File: Data-Science/Ray/Ray-CPU/Matrix_Multiplication/README.md
Comment (Manish-2004) [reply Data-Science/Ray/Ray-CPU/Matrix_Multiplication/README.md:6]: We can use any notebook image

Comment (Manish-2004) [reply Data-Science/Ray/Ray-CPU/Matrix_Multiplication/README.md]: Changed to Data Science

**Comment by sercanCyberVision** [comment]
Thank you @Manish-2004!

@ask664, we are good to merge.

@rashminau, please note that we have replaced Fibonacci example with Matrix_Multiplication. The purpose remains same; showing that Ray performs better under heavy workload. The reason of the change is that Fibonacci did not as expected with `JobSubmissionClient`.

**Pull request merged**
